src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -689,49 +863,202 @@ private static IEnumerable<double> LazySequence(double records, double start, do
             }
         }
 
-        public static FormulaValue Abs(IRContext irContext, NumberValue[] args)
+        private static IEnumerable<decimal> LazySequenceDecimal(int records, decimal start, decimal step)
+        {
+            var x = start;
+            for (var i = 1; i <= records; i++)
+            {
+                yield return x;
+                x += step;
+            }
+        }
+
+        public static FormulaValue Abs(IRContext irContext, FormulaValue[] args)
         {
             var arg0 = args[0];
-            var x = arg0.Value;
-            var val = Math.Abs(x);
+
+            if (arg0 is NumberValue num)
+            {
+                return AbsFloat(irContext, num);
+            }
+            else if (arg0 is DecimalValue dec)
+            {
+                return AbsDecimal(irContext, dec);
+            }
+            else
+            {
+                return CommonErrors.UnreachableCodeError(irContext);
+            }

```

Comment:
I'd write

return arg[0] switch
{
  NumberValue num => AbsFloat(irContext, num),
  DecimalValue dec => AbsDecimal(irContext, dec),
  _ => CommonError....
}

See [https://learn.microsoft.com/en-US/dotnet/csharp/language-reference/operators/switch-expression](https://learn.microsoft.com/en-US/dotnet/csharp/language-reference/operators/switch-expression)

Reply:
I like that!  Thanks!